evaluations/rollups.md

```diff
@@ -0,0 +1,219 @@
+Ethereum L2 Rollups
+===================
+
+A quick and dirty overview of existing rollups and their suitability for hosting
+the Codex marketplace smart contracts. To interact with these contracts, the
+participants in the network create blockchain transactions for purchasing and
+selling storage, and for providing storage proofs that are then checked
+on-chain. It would be too costly for these transactions to happen on Ethereum
+main net, which is why this document explores L2 rollups as an alternative.
+
+Main sources used:
+- individual websites of the rollup projects
+- https://l2beat.com
+- https://blog.kroma.network/l2-scaling-landscape-fees-and-max-tps-fe6087d3f690
+
+Requirements
+------------
+
+For the storage marketplace to work, we have the following requirements for a
+rollup:
+1. Low gas costs; if gas is too costly then the business case of storage
+   providers disappears
+2. EVM compatibility; this shortens our time to market because we already have
+   Solidity contracts
+3. Support for BN254 elliptic curve precompiles (ecAdd, ecMul, ecPairing) for
+   the proof system
+4. High throughput; our current proof system that checks all proofs separately
+   on chain requires a large number of transactions per second
+5. Censorship resistant; an L2 operator should not have the power to exclude
```

Comment:
This is a nice to have, but not an absolute requirement given the current state of rollup ecosystems in general. It is well understood that right now, most of this projects are taking a pragmatic approach and so are we.

Reply:
I agree; the state of the ecosystem doesn't give us much choice.